Project/Project/Services/MovieApiService.cs

```diff
@@ -17,6 +17,58 @@ public MovieApiService(HttpClient httpClient, IConfiguration configuration)
             _apiKey = configuration.GetValue<string>("Api:MovieApi");
         }
 
+        public async Task<string> GetTrailerByImdb(string id)
```

Comment:
Please check if you could imprive this methode with following princpals

- Early return: Avoid unnecessary work by checking conditions and returning early if possible.
- Efficient error handling: Refactor try-catch to avoid unnecessary nesting.